system/jlib/jlog.hpp

```diff
@@ -1342,4 +1344,287 @@ interface IComponentLogFileCreator : extends IInterface
+    }
+
+    void setEnd(const CDateTime  end)
+    {
+        endt = end;
+    }
+
+    void setStart(const char * start, bool local = true)
+    {
+        startt.setString(start,nullptr,local);
+    }
+
+    void setEnd(const char * end, bool local = true)
+    {
+        endt.setString(end,nullptr,local);
+    }
+};
+
+typedef enum
+{
+    LOGACCESS_FILTER_jobid,
+    LOGACCESS_FILTER_class,
+    LOGACCESS_FILTER_audience,
+    LOGACCESS_FILTER_component,
+    LOGACCESS_FILTER_or,
+    LOGACCESS_FILTER_and,
+    LOGACCESS_FILTER_wildcard,
+    LOGACCESS_FILTER_unknown
+} LogAccessFilterType;
+
+inline const char * LogAccessFilterTypeToString(LogAccessFilterType field)
+{
+    switch(field)
+    {
+    case LOGACCESS_FILTER_jobid:
+        return("jobid");
+    case LOGACCESS_FILTER_class:
+        return("class");
+    case LOGACCESS_FILTER_audience:
+        return("audience");
+    case LOGACCESS_FILTER_component:
+        return("component");
+    case LOGACCESS_FILTER_or:
+        return("or");
+    case LOGACCESS_FILTER_and:
+        return("and");
+    case LOGACCESS_FILTER_wildcard:
+         return("*");
+    default:
+        return("UNKNOWN");
+    }
+}
+
+inline unsigned LogAccessFilterTypeFromName(char const * name)
+{
+    if (!name || !*name)
+        return LOGACCESS_FILTER_unknown;
+
+    if(strnicmp(name, "jobid", 5)==0)
+        return LOGACCESS_FILTER_jobid;
+    if(strnicmp(name, "class", 5)==0)
+        return LOGACCESS_FILTER_class;
+    if(strnicmp(name, "audience", 8)==0)
+        return LOGACCESS_FILTER_audience;
+    if(strnicmp(name, "component", 9)==0)
+        return LOGACCESS_FILTER_component;
+    if(strnicmp(name, "or", 2)==0)
+        return LOGACCESS_FILTER_or;
+    if(strnicmp(name, "and", 3)==0)
+        return LOGACCESS_FILTER_and;
+    return LOGACCESS_FILTER_unknown;
+}
+
+interface jlib_decl ILogAccessFilter : public IInterface
+{
+ public:
+    virtual void addToPTree(IPropertyTree * tree) const = 0;
+    virtual void toString(StringBuffer & out) const = 0;
+    virtual LogAccessFilterType filterType() const = 0;
+};
+
+class CLogAccessFilter : implements ILogAccessFilter, public CInterface
+{
+public:
+    IMPLEMENT_IINTERFACE;
+};
+
+extern jlib_decl ILogAccessFilter * getLogAccessFilterFromPTree(IPropertyTree * tree);
+extern jlib_decl ILogAccessFilter * getWUIDLogAccessFilter(const char * wuid);
+extern jlib_decl ILogAccessFilter * getComponentLogAccessFilter(const char * component);
+extern jlib_decl ILogAccessFilter * getAudienceLogAccessFilter(MessageAudience audience);
+extern jlib_decl ILogAccessFilter * getClassLogAccessFilter(LogMsgClass logclass);
+extern jlib_decl ILogAccessFilter * getBinaryLogAccessFilter(ILogAccessFilter * arg1, ILogAccessFilter * arg2, LogAccessFilterType type);
+extern jlib_decl ILogAccessFilter * getBinaryLogAccessFilterOwn(ILogAccessFilter * arg1, ILogAccessFilter * arg2, LogAccessFilterType type);
+extern jlib_decl ILogAccessFilter * getWildCardLogAccessFilter();
```

Comment:
Although very similar in definition and purpose, I can see why ILogAccessFilter and the associated helpers (which mirror the existing ILogMsgFilter implementations) have been introduced for the log access framework.

The jlog interface and implementations are close to what's needed, but come with extra baggage, in particular I think the main thing is that the ILogMsgFilter contains:
```
    virtual bool              mayIncludeCategory(const LogMsgCategory & cat) const = 0;
    virtual unsigned          queryAudienceMask() const = 0;
    virtual unsigned          queryClassMask() const = 0;
    virtual LogMsgDetail      queryMaxDetail() const = 0;
    virtual bool              isCategoryFilter() const { return false; }
```

which are too specific, aren't needed or wanted for the log access side of things.
I am not sure why it was implemented that way originally, rather than having class/audience/detail as a particularly flavour of a filter implementation (I can partially see because of way it's being called by jlog, but think it would have made more sense to abstract those out).

I think the addition of ILogAccessFilter , the implementations etc. are okay at least for now, pragmatically it is probably too much effort to refactor the existing jlog interfaces and implementations such that they were suitable for log access purposes. We may want to revisit if we're really bored some day to unify them.

Reply:
Agreed.